### PR TITLE
DHFPROD-7302: Better messages for common write errors

### PR DESCRIPTION
**Pull request by ryanjdew**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs [hunk 1]
```diff
@@ -728,8 +728,38 @@ function writeContentArray(contentArray, databaseName) {
...
+}
 
-  return fn.head(xdmp.invoke('/data-hub/5/impl/hub-utils/invoke-queue-write.sjs', vars, options));
+function handleWriteErrors(error, contentArray) {
+  switch (error.name) {
+    case 'XDMP-CONFLICTINGUPDATES':
+      let data = error.data[0];
+      let parseUriRegex = /^xdmp\.documentInsert\("([^"]+)".*$/
+      let uri = parseUriRegex.test(data) ? data.replace(parseUriRegex, '$1'): null;
+      throw new Error(`Attempted to right to the same URI multiple times in the same transaction. ${ uri ? 'URI: ' + uri : ''}`);
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs:744]: right -> write

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/impl/flow-utils/writeContentArray.sjs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const flowUtils = require("/data-hub/5/impl/flow-utils.sjs");
+const test = require("/test/test-helper.xqy");
+
+const results = [];
+try {
+    flowUtils.writeContentArray([{uri: "/0.json", value: {}},{uri: "/1.json", value: {}},{uri: "/1.json", value: {}},{uri: "/2.json", value: {}}]);
+    test.assertTrue(false,'Should have thrown error for duplicate URIs!');
+} catch (e) {
+    results.push(test.assertEqual('Attempted to right to the same URI multiple times in the same transaction. URI: /1.json',e.message));
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/impl/flow-utils/writeContentArray.sjs:11]: right -> write

**Review by rjrudin — CHANGES_REQUESTED** [review]
Looks good, just a couple typos

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs (hunk 1 above)
Comment (ryanjdew) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs:744]: 🤦

**Review by srinathgit — APPROVED** [review]

**Review by rjrudin — APPROVED** [review]

**Pull request merged**